www/js/lib/zimfile.js

```diff
@@ -282,6 +285,7 @@ define(['xzdec_wrapper', 'zstddec_wrapper', 'util', 'utf8', 'q', 'zimDirEntry'],
                 var urlPtrPos = readInt(header, 32, 8);
                 return readMimetypeMap(fileArray[0], mimeListPos, urlPtrPos).then(function (data) {
                     var zf = new ZIMFile(fileArray);
+                    zf.name = fileArray[0].name.replace(/(\.zim)[a-z]{2}$/i, '$1');
```

Comment:
Add comment above to explain what this line does: provides an abstract filename, turning e.g. `en_wikipedia.zimaa` into `en_wikipedia.zim` so that the cache code doesn't have to deal with individual split ZIM files, but just sees one abstracted ZIM archive and uses absolute offsets from the start of the abstracted archive.

(This line will leave non-split filenames alone.)